tests/unit/amo/components/TestCollection.js

```diff
@@ -787,6 +788,45 @@ describe(__filename, () => {
     expect(root.find(MetadataCard)).toHaveLength(0);
   });
 
+  it('dispatches deleteCollectionAddon when removeAddon is called', () => {
```

Comment:
Naming is a bit weird, I think we should stick to `remove` everywhere, because that's the user's action anyway.